app/src/main/java/com/portfolio/tasky/entry/EntryActivity.kt

```diff
@@ -2,28 +2,75 @@ package com.portfolio.tasky.entry
 
 
 import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import android.view.animation.TranslateAnimation
 import androidx.appcompat.app.AppCompatActivity
-import androidx.appcompat.widget.Toolbar
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
 import androidx.databinding.DataBindingUtil
 import com.portfolio.tasky.*
 import com.portfolio.tasky.databinding.ActivityEntryBinding
 import com.portfolio.tasky.entry.fragments.LoginFragment
-import com.portfolio.tasky.usecases.FragmentInflater
-import com.portfolio.tasky.usecases.FragmentInflaterImpl
-import com.portfolio.tasky.usecases.ToolbarHandler
-import com.portfolio.tasky.usecases.ToolbarHandlerImpl
+import com.portfolio.tasky.globals.Constants
+import com.portfolio.tasky.usecases.*
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class EntryActivity : AppCompatActivity(), FragmentInflater by FragmentInflaterImpl(), ToolbarHandler by ToolbarHandlerImpl() {
-    private var viewBinding: ActivityEntryBinding? = null
+    private lateinit var viewBinding: ActivityEntryBinding
+    private lateinit var networkChangeReceiver : NetworkChangeReceiver
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         viewBinding = DataBindingUtil.setContentView(this, R.layout.activity_entry)
 
+        setObservers()
         startLoginFragment()
     }
 
+    private fun setObservers() {
+        networkChangeReceiver = NetworkChangeReceiver(this)
+        networkChangeReceiver.observe(this){
+            when(it){
+                NetworkStatus.Available -> networkAvailability(true)
+                NetworkStatus.Unavailable -> networkAvailability(false)
+            }
+        }
+    }
```

Comment:
I'd keep the activities and fragments as dumb as possible. The only thing they should really do is observe states from a ViewModel and update the Views based on that (and things like navigation, permission handling or similar). 

So in this case, I'd let the ViewModel observe the network state and just forward the state to the UI which can then decide what it does with it.

Reply:
I have implemented EntryViewModel and currently observing network change from there. Only updating views in Activity class like setting text, animating and changing visibility of views.

Earlier also the NetworkChangeReceiver was extended to LiveData<NetworkStatus>, and I was observing it by initialising it in the activity..I thought we can avoid creating an extra ViewModel since the NetworkChangeReceiver was being observed.